docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,231 @@
+
+This object will contain the pointing sky coordinates, the GTI, and the setup of all the 
+models (spectra, spatial morphology, temporal model) for any given source, and it is passed 
+as input parameter to the `MapDatasetEventSampler`. It is important to note that the 
+`MapDataset` object can store information for more than one source. Then, a `.sample` method 
+will draw the sampled events and will provide an output `~astropy.table.Table` object. The 
+latter will containt the reconstructed sky positions, energies, times, and an `EVENT_ID` and 
+`MC_ID`. `EVENT_ID` is a unique number or a string to identify the sampled event, while 
+`MC_ID` is a unique ID (number or string) to identify the model component the event was sampled
+from. The `MapDatasetSampler` should also fill the mandatory header information for event list 
+files decribed on `gamma-astro-data-formats`_. The `MapDatasetSampler` will be designed to work 
+alternatively for input flux or counts maps.
+
+With the code example above, this could go as:
+
+::
+    
+    dataset = MapDataset([pointing, GTI, models])
+    dataset_sampler = MapDatasetEventSampler(dataset, random_state)
+    event_list = dataset_sampler.sample()
+    
+where `random_state` is the seed for the random number generator. 
+
+
+MapDatasetEventSampler
+----------------------
+
+The general design of the `sample` method is as follows:
+
+::
+
+    def sample(dataset, random_state)
+       """Sample events from a `MapDataset`"""
+
+       events_list = []
+    
+       for evaluator in dataset.evaluators:
+            npred = evaluator.compute_npred()
+            n_events = random_state.poisson(npred.data.sum())
+            events = npred.sample(n_events, random_state)
+            events_list.append(events)
+    
+       events_src = vstack(events_list)
+       events_src = dataset.psf.distribute(events_src, random_state)
+       events_src = dataset.edisp.distribute(events_src, random_state)
+    
+       n_events_bkg = random_state.poisson(dataset.background_model.map.data.sum())
+       events_bkg = dataset.background_model.sample(n_events, random_state)
+    
+       events_total = vstack([events_src, events_bkg])
+       events_total.meta = get_events_meta_data(dataset)
+       return EventList(events_total)
+    
+
+In more detail, `sample` starts a loop over the sources stored into the `MapDataset` 
+object. Then, for each source, the `src.evaluate` method will calculate the predicted 
+number of source counts, that will be the input of the `npred_total` method. The latter 
+uses a Poisson distribution, with mean equal to the predicted counts, to estimate the
+random number of sampled events.
+
+We propose to add a `Map.sample(n_events=, random_state=)` method in `~gamampy.cube.MapDataset` 
+that will be the core of the sampling process. The `sample` is based on the 
+`~gammapy.utils.random.InverseCDFSampler` class described in `PR#2229`_ . The output 
```

Comment:
Please here and everywhere when referencing classes, just write this:
```
``InverseCDFSampler``
```
The single-backticks and ~ are Sphinx-syntax to create links to API docs. But if the API changes in the coming years, we don't want to have to go back and update PIGs. So from PIGs we don't create these links.